### DISCUSSION TITLE
Column-specific filters in the tool-swapper

### DISCUSSION THREAD

**Discussion by Pratixx** [P3pp3rF1y/SophisticatedBackpacks#1089] [Feature / Idea Archive]
The current tool-swapper does not work well if you have both Fortune and Silk Touch tools, so I believe it would be a nice addition to have all tools be able to be filtered into groupings, and if the tool currently equipped is part of that grouping, only a tool in the same grouping as the one being held will be selected to swap to. For example, you could have a Silk Touch group and a Fortune group (named here for simplicity) with their respective tools in them (Silk Touch shovel, pick, axe in Silk Touch category, Fortune shovel, pick, axe in Fortune category), where only tools of the same group as the one being held will be selected. This will negate the issue of switching to a Silk Touch pick when mining ore or a Fortune shovel when digging grass blocks.

**Comment by P3pp3rF1y** [comment]
This seems very specific to how you would probably play with it, but at least myself I don't usually have fortune or silktouch on anything other than pickaxe so this would be of no benefit to me and the players I watch they don't usually have these enchantments on anything else either unless they really need them to get some specific blocks. 
And if I were to tackle somehow considering enchantments in tool swapper I would like that logic to work for people who want to use one pickaxe for a set of blocks while another pickaxe for another set and this logic would also need to be univeral enough to work with modded enchantments like auto smelt, pulverizing, vein mining, ...
